paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/samples/ui/customersheet/playground/CustomerSheetPlaygroundViewModel.kt

```diff
@@ -0,0 +1,329 @@
+        }
+    }
+
+    fun onCustomerSheetResult(result: CustomerSheetResult) {
+        when (result) {
+            is CustomerSheetResult.Canceled -> {
+                updateViewState<CustomerSheetPlaygroundViewState.Data> {
+                    it.copy(
+                        selection = result.selection,
+                        errorMessage = null,
+                    )
+                }
+            }
+            is CustomerSheetResult.Selected -> {
+                updateViewState<CustomerSheetPlaygroundViewState.Data> {
+                    it.copy(
+                        selection = result.selection,
+                        errorMessage = null,
+                    )
+                }
+            }
+            is CustomerSheetResult.Error -> {
+                updateViewState<CustomerSheetPlaygroundViewState.Data> {
+                    it.copy(
+                        selection = null,
+                        errorMessage = result.exception.message,
+                    )
+                }
+            }
+        }
+    }
+
+    private fun toggleSetupIntentEnabled() {
+        updateViewState<CustomerSheetPlaygroundViewState.Data> {
+            it.copy(
+                isSetupIntentEnabled = !it.isSetupIntentEnabled
+            )
+        }
+    }
+
+    private fun toggleGooglePayEnabled() {
+        updateViewState<CustomerSheetPlaygroundViewState.Data> {
+            it.copy(
+                isGooglePayEnabled = !it.isGooglePayEnabled
+            )
+        }
+    }
+
+    private fun toggleExistingCustomer() {
+        val isExistingCustomer = !viewState.value.isExistingCustomer
+        updateViewState<CustomerSheetPlaygroundViewState.Data> {
+            it.copy(
+                isExistingCustomer = isExistingCustomer,
+            )
+        }
+
+        viewModelScope.launch {
+            when (val result = fetchCustomerEphemeralKey(isExistingCustomer)) {
+                is Result.Success -> {
+                    PaymentConfiguration.init(
+                        context = getApplication(),
+                        publishableKey = result.value.publishableKey,
+                    )
+                }
+                is Result.Failure -> {
+                    _viewState.update {
+                        CustomerSheetPlaygroundViewState.FailedToLoad(
+                            message = result.error.message.toString()
+                        )
+                    }
+                }
+            }
+        }
+    }
+
+    private inline fun <reified T : CustomerSheetPlaygroundViewState> updateViewState(block: (T) -> T) {
+        (_viewState.value as? T)?.let {
+            _viewState.update {
+                block(it as T)
+            }
+        }
+    }
+
+    class Factory(
+        private val application: Application,
+    ) : ViewModelProvider.Factory {
+        @Suppress("UNCHECKED_CAST")
+        override fun <T : ViewModel> create(modelClass: Class<T>): T {
```

Comment:
Nit: Can use `create(Class<T>, CreationExtras)` to retrieve the `Application` instead of having to provide it to the `Factory`.